linkerd/server-policy/src/meta.rs

```diff
@@ -16,31 +16,45 @@ pub enum Meta {
 
 impl Meta {
     pub fn new_default(name: impl Into<Cow<'static, str>>) -> Arc<Self> {
-        Arc::new(Meta::Default { name: name.into() })
+        Arc::new(Self::Default { name: name.into() })
     }
 
     pub fn name(&self) -> &str {
         match self {
-            Meta::Default { name } => name,
-            Meta::Resource { name, .. } => name,
+            Self::Default { name } => name,
+            Self::Resource { name, .. } => name,
         }
     }
 
     pub fn kind(&self) -> &str {
         match self {
-            Meta::Default { .. } => "default",
-            Meta::Resource { kind, .. } => kind,
+            Self::Default { .. } => "default",
+            Self::Resource { kind, .. } => kind,
         }
     }
 
     pub fn group(&self) -> &str {
         match self {
-            Meta::Default { .. } => "",
-            Meta::Resource { group, .. } => group,
+            Self::Default { .. } => "",
+            Self::Resource { group, .. } => group,
         }
     }
 }
 
+impl std::cmp::PartialEq for Meta {
```

Comment:
nit: it might be nice if there was a comment explaining why this is manually implemented?